File: core/src/rustyice/game/tiles/FloorTile.java
package rustyice.game.tiles;

import com.badlogic.gdx.graphics.Color;
import com.badlogic.gdx.graphics.g2d.SpriteBatch;
import rustyice.graphics.Camera;
import rustyice.graphics.RenderFlags;

public class FloorTile extends Tile {

    @Override
    public void update(float delta) {

    }

    @Override
    public void render(SpriteBatch batch, Camera camera, int flags) {
        if((flags & RenderFlags.NORMAL) == RenderFlags.NORMAL){
            batch.setColor(Color.WHITE);
            batch.draw(getResources().box, getX(), getY(), TileMap.TILE_SIZE, TileMap.TILE_SIZE);
        }
    }

    @Override
    public void init() {
        super.init();
    }
}

File: core/src/rustyice/game/tiles/FloorTile.kt
package rustyice.game.tiles

import com.badlogic.gdx.graphics.Color
import com.badlogic.gdx.graphics.g2d.Sprite
import rustyengine.RustyEngine
import rustyice.game.tiles.Tile
import rustyengine.resources.Resources

class FloorTile(): Tile() {
    override fun init(){
        super.init()

        val sprite = Sprite(RustyEngine.resorces.box)
        sprite.color = Color.CORAL

        this.sprite = sprite
    }
}
